compiler/rustc_const_eval/src/interpret/operand.rs

```diff
@@ -300,13 +318,18 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
     /// Note that for a given layout, this operation will either always fail or always
     /// succeed!  Whether it succeeds depends on whether the layout can be represented
     /// in an `Immediate`, not on which data is stored there currently.
+    ///
+    /// If `force` is `true`, then even scalars with fields that can be ununit will be
+    /// read. This means the load is lossy and should not be written back!
+    /// This flag exists only for validity checking.
     pub fn try_read_immediate(
         &self,
         src: &OpTy<'tcx, M::PointerTag>,
+        force: bool,
```

Comment:
Considering most users use and should use `false` maybe make this one private and expose two wrapper functions (where the old name uses false) and give the one for `true` a more descriptive name

Reply:
Really `try_read_immediate` should be private, but sadly const-prop calls it. I wonder if there is a way around that...

Given that this is an internal API exposed only for const-prop, an alternative might be to give it a more scary name and keep the `force` parameter.

Reply:
> sadly const-prop calls it. I wonder if there is a way around that...

😆 we're doing too much multi-track drifting around const prop discussions. This will obviously go away if we limit const prop to cast.rs and operator.rs

Scary name sgtm